docs/getting_started.md

```diff
@@ -0,0 +1,173 @@
+3. Install [pip](https://pip.pypa.io/en/stable/installation/). This is the Python package manager we use.
+4. Install the VSCode [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) extension (optional but recommended). 
+5. Install the VSCode [HashiCorp Terraform](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform) extension (optional but recommended).
+ 
+
+### Developing Python Google Cloud Functions
+
+At a high level, we follow the guide [here](https://cloud.google.com/functions/docs/how-to) for developing Python runtime Cloud Functions. Please note that this guide also provides documentation for Cloud Functions using other runtimes beyond Python so make sure to read carefully.
+
+#### Cloud Function Directory Structure
+
+All Cloud Functions live in the [cloud-functions](https://github.com/CDCgov/phdi-google-cloud/tree/main/cloud-functions) directory. The tree below shows a hypoethetical example for a Cloud Function called `myfunction`. For Python Cloud Functions GCP requires that each function have a dedicated directory containing a `main.py` files with the function's entry point along with a `requirements.txt` specifying all of the function's dependencies. The PHDI team believes strongly in the importance of developing well tested code so we have chosen to include and additional with the the name `test_<FUNCTION-NAME>.py`, in this case `test_myfunction.py`, that cotains unit tests for the function. The deployment process for `myfunction` simply passes a zip file of the entire directory to GCP.
+
+```bash
+cloud-functions/
+├── requirements_dev.txt
+└── myfunction/
+    ├── main.py
+    ├── requirements.txt
+    └── test_myfunction.py
+```
+
+#### Creating a Virtual Environment
+
+In order to avoid dependency conflicts between multiple Python projects and potentially between different Cloud Functions within this repo, we recommend that all Cloud Function development is done within a Python virtual environment dedicated to a single function. For information on creating, activating, deactivating, and managing Python virtual environment please refer to [this guide](https://realpython.com/python-virtual-environments-a-primer). We recommend naming your virtual environment `.venv` as we have already added to our `.gitignore` to prevent it from being checked into source control.
+
+#### Cloud Function Dependencies
+
+After creating a virtual environment and activating it you may install all of the Cloud Function's dependencies from its root directory with `pip install -r requirements.txt`. To create or update a `requirements.txt` files run `pip freeze > requirements.txt`. Please not that all Cloud Functions require the [Functions Framework](https://cloud.google.com/functions/docs/functions-framework) which can be installed with `pip install functions-framework`.
+
```

Comment:
Do you have an example of a Cloud Function that wouldn't require Functions Framework? I think at least the entry point with require an `@functions_framework`. Maybe I'm missing something though.

Reply:
I was just using the text that was already there and thought it was saying that not all cloud functions require it.  But I agree it would make sense that they all would require the GCP SDK.  Maybe the text could be "All Cloud Functions require the Functions Framework which can be installed with...."

Reply:
Okay I believe this has been corrected now.